aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -2869,8 +2888,263 @@ impl AptosVM {
                 )?;
                 (vm_status, output)
             },
+            Transaction::ScheduledTransaction(txn) => {
+                let traversal_storage = TraversalStorage::new();
+                let mut traversal_context = TraversalContext::new(&traversal_storage);
+                let (vm_status, output) = self.process_scheduled_transaction(
+                    resolver,
+                    txn,
+                    &mut traversal_context,
+                    code_storage,
+                    log_context,
+                )?;
+                (vm_status, output)
+            },
         })
     }
+
+    pub(crate) fn process_scheduled_transaction(
+        &self,
+        resolver: &impl AptosMoveResolver,
+        txn: &ScheduledTransactionInfoWithKey,
+        traversal_context: &mut TraversalContext,
+        code_storage: &(impl AptosCodeStorage + BlockSynchronizationKillSwitch),
+        log_context: &AdapterLogSchema,
+    ) -> Result<(VMStatus, VMOutput), VMStatus> {
+        // 'base_gas_amount' is gas used for prologue, epilogue and for getting ready txns from
+        // schedule queue
+        let balance = txn.max_gas_amount - SCHEDULED_TRANSACTIONS_MODULE_INFO.base_gas_amount;
+        let mut system_txns_gas_meter = UnmeteredGasMeter;
+
+        let storage_gas = self.storage_gas_params(log_context)?;
+        let mut gas_meter = make_prod_gas_meter(
```

Comment:
nit: move this down helps with readability, not to confuse with the system_gas_meter